### PR TITLE
polish delete database modal

### PR DESCRIPTION
**Pull request by alxnddr**
## Changes

1. Adds missing loading state to the delete database modal
2. Adds a softer animation when a user confirms the removal of all content

## Demo

<video src="https://user-images.githubusercontent.com/14301985/216742720-7a2f1149-dfca-474d-b874-a2d1d2389db1.mov" />

## How to verify 

- Admin -> Databases
- Open your sample database and copy the connection string
- Connect a new H2 database using the connection string
- Create one question based on the data from db
- Delete the database and ensure the modal has animation once you confirm the removal of the question

### CODE REVIEW

**Review by mazameli — APPROVED** [review]
Thank you!

File: frontend/src/metabase/admin/databases/components/DeleteDatabaseModel/DeleteDatabaseModal.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React, { FormEvent, useState } from "react";
+import React, { FormEvent, useEffect, useState } from "react";
```
Comment (npfitz) [inline frontend/src/metabase/admin/databases/components/DeleteDatabaseModel/DeleteDatabaseModal.tsx:1]: does `useEffect` get used in the component?

**Review by npfitz — APPROVED** [review]
Aside from that one minor comment, all looks good to me

File: frontend/src/metabase/admin/databases/components/DeleteDatabaseModel/DeleteDatabaseModal.unit.spec.tsx [hunk 2]
```diff
@@ -130,4 +130,15 @@ describe("DeleteDatabaseModal", () => {
 
     expect(screen.getByText("Something went wrong")).toBeInTheDocument();
   });
+
+  it("shows a loader while fetching usage info", () => {
```
Comment (gusaiani) [inline frontend/src/metabase/admin/databases/components/DeleteDatabaseModel/DeleteDatabaseModal.unit.spec.tsx:134]: 👍

**Review by gusaiani — APPROVED** [review]
Looking good. 

Just that unused `useEffect` that linter will pick up.

**Pull request merged**